lib/gssapi/mech/gss_set_cred_option.c

```diff
@@ -41,7 +41,7 @@ gss_set_cred_option (OM_uint32 *minor_status,
 	struct _gss_cred *cred = (struct _gss_cred *) *cred_handle;
 	OM_uint32	major_status = GSS_S_COMPLETE;
 	struct _gss_mechanism_cred *mc;
-	int one_ok = 0;
+	OM_uint32 one_ok = 0;
```

Comment:
one_ok is not a protocol variable.  Its a local variable used as a boolean flag.  Its proper type is 'int'.

Reply:
Thanks, I agree.  I can't see how this was ever an issue, I'll drop that hunk.

Reply:
Looping back to this one, there is no krb5_boolean available in gsspai, is Heimdal willing to use C99 bool yet?

The issue is that on Ubuntu 14.04 with strict compiler options and -O3 we get:


==> /builds/catalyst-samba/samba/samba-o3.stderr <==
../source4/heimdal/lib/gssapi/mech/gss_set_cred_option.c: In function ‘gss_set_cred_option’:
../source4/heimdal/lib/gssapi/mech/gss_set_cred_option.c:112:5: error: assuming signed overflow does not occur when simplifying conditional to constant [-Werror=strict-overflow]
  if (one_ok) {
     ^
cc1: some warnings being treated as errors

If I put that back into the commit message, can I have this hunk accepted?

Reply:
This warning has to be a compiler bug.    
I'm curious, what happens if you change that to 

   if (!one_ok)
       return major_status;

   *minor_status = 0;
   return GSS_S_COMPLETE;